src/devices/sound/k054539.cpp

```diff
@@ -299,10 +339,13 @@ void k054539_device::sound_stream_update(sound_stream &stream, std::vector<read_
 					base1[0x0d] = cur_pos>> 8 & 0xff;
 					base1[0x0e] = cur_pos>>16 & 0xff;
 				}
+			} else {
+				// Fill the interpolation vectors with silence when channel is disabled
+				advance_filter(ch, 0);
 			}
 		reverb_pos = (reverb_pos + 1) & 0x1fff;
-		outputs[0].put_int(sample, lval, 32768);
-		outputs[1].put_int(sample, rval, 32768);
+		outputs[0].put_int(sample, lval * 32768.0f, 32768);
+		outputs[1].put_int(sample, rval * 32768.0f, 32768);
```

Comment:
Aren’t your `lval` and `rval` already correctly scaled `float` values?  Can’t you just do this?
```
        outputs[0].put(sample, lval);
        outputs[1].put(sample, rval);
```